app/src/main/java/nu/yona/app/api/manager/impl/NotificationManagerImpl.java

```diff
@@ -161,6 +162,87 @@ public void onError(Object errorMessage) {
         }
     }
 
+
+    public void getMessageWithUrl(final String urlForMessagesFetch, final boolean isUnreadStatus, final DataLoadListener listener, final boolean isProcessUpdate) {
+        try {
+            User user = YonaApplication.getEventChangeManager().getDataState().getUser();
+            if (user != null && user.getLinks() != null && user.getLinks().getYonaMessages() != null
+                    && !TextUtils.isEmpty(user.getLinks().getYonaMessages().getHref())) {
+
+                notificationNetwork.getNextSetOfMessagesFromURL(urlForMessagesFetch, YonaApplication.getEventChangeManager().getSharedPreference().getYonaPassword(), isUnreadStatus, new DataLoadListener() {
+                    @Override
+                    public void onDataLoad(Object result) {
+                        if (listener != null) {
+                            if (result instanceof YonaMessages) {
+                                YonaMessages resultYonaMessagesObj = (YonaMessages) result;
+                                if (resultYonaMessagesObj != null && resultYonaMessagesObj.getEmbedded() != null) {
+                                    Embedded embedded = resultYonaMessagesObj.getEmbedded();
+                                    List<YonaMessage> listMessages = embedded.getYonaMessages();
+                                    boolean isAnyProcessed = false;
+                                    SimpleDateFormat sdf = new SimpleDateFormat(AppConstant.YONA_LONG_DATE_FORMAT, Locale.getDefault());
```

Comment:
This is declared 20 lines before its first usage. Move it down to where you need it.
The likely reason you did this is that you don't want to instantiate the ``SimpleDateFormat`` in the loop. To prevent that, switch from ``java.util.Date`` to the date/time classes in ``java.time`` and its subpackages.

Reply:
I will update code here.

Reply:
OK

Reply:
java.time package requires minimum Android API level of 19.

Reply:
Good to hear. That's our minimum, right?